src/main/java/Duke.java

```diff
@@ -1,10 +1,258 @@
+                if (taskDetails[0].equals("D")) {
+                    loadDeadline(i,taskDetails[2]);
+                    list[i].isDone = Boolean.parseBoolean(taskDetails[1]);
+                }
+            }
+        }
+    }
+
+    public static void saveFileContents(String filePath) {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            // Write inputCount to first line of file
+            fw.write(Integer.toString(inputCount));
+            fw.write(System.lineSeparator());
+            for(int i = 0; i < inputCount; i++) {
+                fw.write(toSave(list[i]));
+                fw.write(System.lineSeparator());
+            }
+            fw.close();
+            System.out.println("File saved successfully.");
+        } catch (IOException e) {
+            System.out.println("Error: File not found! (IOException)");
+        }
+    }
+
+    public static void main(String[] args) {
+        // Load file
+        try {
+            loadFileContents(filePath);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found!");
+        }
+        printGreeting();
+
+        Scanner sc = new Scanner(System.in);
+        while(true) {
+            String input = sc.next();
+            // Exit Command
+            if(input.equals("bye")) {
+                saveFileContents(filePath);
+                System.out.println("Bye. Hope to see you again soon!");
+                break;
+            }
+            else {
+                // Check for other commands
+                switch(input) {
+                case ("list"):
+                    listTasks();
+                    break;
+
+                case ("done"):
+                    String done = sc.nextLine();
+                    doneTask(done);
+                    break;
+
+                case ("delete"):
+                    String delete = sc.nextLine();
+                    deleteTask(delete);
+                    break;
+
+                case ("todo"):
+                    String description = sc.nextLine();
+                    // Throw exception if description is empty
+                    try {
+                        if (description.equals("")) {
+                            throw new DukeException();
+                        }
+                        addToDo(description);
+                    } catch (DukeException e) {
+                        System.out.println("Error: Did not specify todo description!");
+                    }
+                    break;
+
+                case ("deadline"):
+                    String desc = sc.nextLine();
+                    try {
+                        if(desc.equals("")) {
+                            throw new DukeException();
+                        }
+                        addDeadline(desc);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        System.out.println("Error: Did not specify deadline correctly!");
```

Comment:
You can add the correct way for specifying deadline here, so 

Error: Did not specify deadline correctly! Please input [correct way].

On top of that, you can consider changing all print statements / exception statements to string constants and declare them all at the top for easy reference and future amendments.